pkg/registry/extensions/podsecuritypolicy/storage/storage.go

```diff
@@ -35,9 +35,9 @@ type REST struct {
 // NewREST returns a RESTStorage object that will work against PodSecurityPolicy objects.
 func NewREST(optsGetter generic.RESTOptionsGetter) *REST {
 	store := &genericregistry.Store{
-		NewFunc:                  func() runtime.Object { return &extensions.PodSecurityPolicy{} },
-		NewListFunc:              func() runtime.Object { return &extensions.PodSecurityPolicyList{} },
-		DefaultQualifiedResource: extensions.Resource("podsecuritypolicies"),
+		NewFunc:                  func() runtime.Object { return &policy.PodSecurityPolicy{} },
+		NewListFunc:              func() runtime.Object { return &policy.PodSecurityPolicyList{} },
+		DefaultQualifiedResource: policy.Resource("podsecuritypolicies"),
```

Comment:
this change is wrong. It's responsible for the KUBE_API_VERSIONS panic.

Reply:
@sttts Thank you, Stefan! I'll look at it closely today, because I don't understand to what I should replace it then. I thought that as we're using internal type here, it also should be updated.

Reply:
referencing policy here makes it look up the resource for the policy group. Take a look at other storages, whether we parametrize this for each group.

Reply:
Looks like the following fixes the issue:
```diff
--- a/pkg/registry/extensions/podsecuritypolicy/storage/storage.go
+++ b/pkg/registry/extensions/podsecuritypolicy/storage/storage.go
@@ -17,6 +17,7 @@ limitations under the License.
 package storage
 
 import (
+       extensions "k8s.io/api/extensions/v1beta1"
        "k8s.io/apimachinery/pkg/runtime"
        "k8s.io/apiserver/pkg/registry/generic"
        genericregistry "k8s.io/apiserver/pkg/registry/generic/registry"
@@ -37,7 +38,7 @@ func NewREST(optsGetter generic.RESTOptionsGetter) *REST {
        store := &genericregistry.Store{
                NewFunc:                  func() runtime.Object { return &policy.PodSecurityPolicy{} },
                NewListFunc:              func() runtime.Object { return &policy.PodSecurityPolicyList{} },
-               DefaultQualifiedResource: policy.Resource("podsecuritypolicies"),
+               DefaultQualifiedResource: extensions.Resource("podsecuritypolicies"),
 
                CreateStrategy:      podsecuritypolicy.Strategy,
                UpdateStrategy:      podsecuritypolicy.Strategy,
```
I'll push an update.

Reply:
@sttts It worked and it's green now! Thank you for your help! If you're ok with result, could you LGTM it?

Reply:
Will it work when disabling the extensions group now, and enabling the policy group? 👺

Reply:
How to check that, run the api server with only policy API group and create PSP?

Reply:
When I run api server only with policy group enabled it panics like it did before:
```shell
$ KUBE_API_VERSIONS=v1,policy/v1beta1 /home/coder/git/src/k8s.io/kubernetes/_output/local/bin/linux/amd64/kube-apiserver --bind-address=127.0.0.1 --storage-backend=etcd3 --etcd-servers=http://127.0.0.1:2379 --etcd-prefix=/randomPrefix --runtime-config=api/all=false,api/v1=true,policy/v1beta1=true  --cert-dir=/tmp/
...
panic: unable to find storage destination for {extensions podsecuritypolicies}, due to group extensions has not been registered

goroutine 1 [running]:
k8s.io/kubernetes/pkg/registry/extensions/podsecuritypolicy/storage.NewREST(0x9a33900, 0xc4202925a0, 0x3da6429)
	/home/coder/git/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/registry/extensions/podsecuritypolicy/storage/storage.go:52 +0x59d
k8s.io/kubernetes/pkg/registry/policy/rest.RESTStorageProvider.v1beta1Storage(0x9a45a80, 0xc4204d6078, 0x9a33900, 0xc4202925a0, 0x7)
	/home/coder/git/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/registry/policy/rest/storage_policy.go:52 +0x14c
k8s.io/kubernetes/pkg/registry/policy/rest.RESTStorageProvider.NewRESTStorage(0x9a45a80, 0xc4204d6078, 0x9a33900, 0xc4202925a0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/coder/git/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/registry/policy/rest/storage_policy.go:39 +0x1dd
k8s.io/kubernetes/pkg/registry/policy/rest.(*RESTStorageProvider).NewRESTStorage(0x9c356e8, 0x9a45a80, 0xc4204d6078, 0x9a33900, 0xc4202925a0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	<autogenerated>:1 +0xeb
k8s.io/kubernetes/pkg/master.(*Master).InstallAPIs(0xc420095d60, 0x9a45a80, 0xc4204d6078, 0x9a33900, 0xc4202925a0, 0xc420e31550, 0xf, 0xf)
...
```

Reply:
I expected that. Would check in other groups with cohabitation how or whether this is solved there. But otherwise, I would leave the extensions group there for the moment. As we understand now what happens, I think it is fine. Maybe add a TODO above that.

Reply:
@sttts Thanks! I'm not fully follow you on what to do next. Could you provide "action items" to me (I don't know what to put to this comment, for example)? If you think that there is nothing to do, could you add LGTM and cancel hold?